Fix pdk validate warnings

avitacco opened the pull request:
This PR fixes all of the warnings from `pdk validate` related to this module. It is mostly documentation and spacing changes, but several code changes had to be made. The main thing is the removal of all of the anchor resources in favor of `require` parameters. This PR also updates the version of PDK used from 3.0 to 3.4.

## Notes for Reviewers

- [ ] The commit history must be preserved - please use the rebase-merge or standard merge option instead of squash-merge
- [ ] Sync up with the author before merging

avitacco committed:
Changed code to fix all issues found by `pdk validate`

Corrected incorrect spacing issues in a couple of files. Added data
types for all parameters for all classes. Added documentation for all
parameters. Removed anchor tags and replaced with require parameters
where appropriate.

avitacco committed:
Merge branch 'Graylog2:main' into fix-pdk-validate-warnings

avitacco committed:
Updating PDK

As part of fixing all PDK warnings, I realized that PDK was warning that
the module needed to be updated to the newer version of PDK.

bernd reviewed (approved):
Thank you!

bernd commented:
@avitacco Hey! Can you fix the conflicts and the test errors? Thank you! :pray:

avitacco committed:
Merge branch 'main' into fix-pdk-validate-warnings

avitacco committed:
Fixing some more issues

Adding a dependency for the package for the configuration file since the
directory doesn't exist unless the package is installed. Also added a
test to ensure that the dependency exists. Finally, changed some stdlib
function calls to use the newer namespaced names.

bernd reviewed (changes requested):
There are some test errors.

avitacco commented:
@bernd I just wanted to give a quick update on this. My work is currently crazy and taking up all of my time. I have not forgotten about this PR or any of my other plans. I don't have the time right now, but hopefully soon.

bernd commented:
> @bernd I just wanted to give a quick update on this. My work is currently crazy and taking up all of my time. I have not forgotten about this PR or any of my other plans. I don't have the time right now, but hopefully soon.

@avitacco No worries, take all the time you need! Thanks for the update. :pray:

avitacco committed:
Changing some whitespace around
